### ISSUE TITLE
Cannot update entries / undefined method getutc from nil:NilClass

### ISSUE THREAD

**Issue by arnaud** [motor-admin/motor-admin#23]
The following error message pops when trying to update some entries:

> undefined method `getutc' from nil:NilClass

Here is the stack trace, in case it would ring a bell to someone:

```
E, [2021-11-25T21:44:41.159110 #7]
ERROR -- : [1a4c0e06-e8b9-471d-b877-47a0ebe10823] undefined method `getutc' for nil:NilClassE, [2021-11-25T21:44:41.162530 #7]
ERROR -- : [1a4c0e06-e8b9-471d-b877-47a0ebe10823] 
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/postgresql/oid/timestamp_with_time_zone.rb:19:in `cast_value'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activemodel-7.0.0.alpha2/lib/active_model/type/value.rb:46:in `cast'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activemodel-7.0.0.alpha2/lib/active_model/type/helpers/accepts_multiparameter_time.rb:16:in `cast'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activemodel-7.0.0.alpha2/lib/active_model/attribute.rb:179:in `type_cast'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activemodel-7.0.0.alpha2/lib/active_model/attribute.rb:43:in `value'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activemodel-7.0.0.alpha2/lib/active_model/attribute_set/builder.rb:43:in `fetch_value'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/transactions.rb:403:in `block in restore_transaction_record_state'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activemodel-7.0.0.alpha2/lib/active_model/attribute_set.rb:93:in `transform_values'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activemodel-7.0.0.alpha2/lib/active_model/attribute_set.rb:93:in `map'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/transactions.rb:402:in `restore_transaction_record_state'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/transactions.rb:334:in `rolledback!'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/abstract/transaction.rb:134:in `rollback_records'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/abstract/transaction.rb:312:in `block in rollback_transaction'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/abstract/transaction.rb:309:in `rollback_transaction'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/abstract/transaction.rb:325:in `rescue in block in within_new_transaction'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/abstract/transaction.rb:322:in `block in within_new_transaction'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `handle_interrupt'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/concurrency/load_interlock_aware_monitor.rb:25:in `block in synchronize'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `handle_interrupt'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/concurrency/load_interlock_aware_monitor.rb:21:in `synchronize'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/abstract/transaction.rb:317:in `within_new_transaction'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/connection_adapters/abstract/database_statements.rb:316:in `transaction'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/transactions.rb:350:in `with_transaction_returning_status'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activerecord-7.0.0.alpha2/lib/active_record/persistence.rb:702:in `update!'
[...]
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/journey/router.rb:32:in `each'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/journey/router.rb:32:in `serve'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/routing/route_set.rb:845:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/railties-7.0.0.alpha2/lib/rails/engine.rb:530:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/railties-7.0.0.alpha2/lib/rails/railtie.rb:207:in `public_send'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/railties-7.0.0.alpha2/lib/rails/railtie.rb:207:in `method_missing'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/routing/mapper.rb:19:in `block in '
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/routing/mapper.rb:48:in `serve'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/journey/router.rb:50:in `block in serve'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/journey/router.rb:32:in `each'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/journey/router.rb:32:in `serve'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/routing/route_set.rb:845:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/warden-1.2.9/lib/warden/manager.rb:36:in `block in call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/warden-1.2.9/lib/warden/manager.rb:34:in `catch'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/warden-1.2.9/lib/warden/manager.rb:34:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/tempfile_reaper.rb:15:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/etag.rb:27:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/conditional_get.rb:40:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/head.rb:12:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/http/permissions_policy.rb:22:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/http/content_security_policy.rb:18:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/session/abstract/id.rb:266:in `context'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/session/abstract/id.rb:260:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/middleware/cookies.rb:693:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/middleware/callbacks.rb:27:in `block in call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/callbacks.rb:99:in `run_callbacks'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/middleware/callbacks.rb:26:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/middleware/debug_exceptions.rb:28:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/middleware/show_exceptions.rb:26:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/lograge-0.11.2/lib/lograge/rails_ext/rack/logger.rb:15:in `call_app'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/railties-7.0.0.alpha2/lib/rails/rack/logger.rb:25:in `block in call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/tagged_logging.rb:99:in `block in tagged'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/tagged_logging.rb:37:in `tagged'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/activesupport-7.0.0.alpha2/lib/active_support/tagged_logging.rb:99:in `tagged'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/railties-7.0.0.alpha2/lib/rails/rack/logger.rb:25:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/middleware/remote_ip.rb:93:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/request_store-1.5.0/lib/request_store/middleware.rb:19:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/middleware/request_id.rb:26:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/method_override.rb:24:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/runtime.rb:22:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/middleware/executor.rb:14:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/middleware/static.rb:24:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/rack-2.2.3/lib/rack/sendfile.rb:110:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/actionpack-7.0.0.alpha2/lib/action_dispatch/middleware/host_authorization.rb:90:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/railties-7.0.0.alpha2/lib/rails/engine.rb:530:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/puma-5.5.2/lib/puma/configuration.rb:249:in `call'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/puma-5.5.2/lib/puma/request.rb:77:in `block in handle_request'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/puma-5.5.2/lib/puma/thread_pool.rb:340:in `with_force_shutdown'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/puma-5.5.2/lib/puma/request.rb:76:in `handle_request'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/puma-5.5.2/lib/puma/server.rb:447:in `process_client'
/__enclose_io_memfs__/lib/ruby/gems/2.7.0/gems/puma-5.5.2/lib/puma/thread_pool.rb:147:in `block in spawn_thread'

I, [2021-11-25T21:44:41.164457 #7]  INFO -- : [1a4c0e06-e8b9-471d-b877-47a0ebe10823] {"method":"PUT","path":"/api/data/users/17602","status":500,"duration":18.78,"view":0.45,"db":2.23,"fwd":"[REDACTED]","uid":1}
```

**Comment by omohokcoj** [comment]
@arnaud can you please provide you table schema as well as the data you submit via motor admin form (you can check request data via browser network inspector tool).

**Comment by arnaud** [comment]
Here is the DDL of said table:

```sql
CREATE TABLE schema.table (
    id bigint GENERATED ALWAYS AS IDENTITY PRIMARY KEY,
    email citext NOT NULL UNIQUE,
    passwd text NOT NULL,
    status user_registration_status NOT NULL DEFAULT 'unconfirmed'::user_registration_status,
    confirmation_token text NOT NULL DEFAULT md5(random()::text || clock_timestamp()::text) UNIQUE,
    confirmed_at timestamp with time zone,
    first_name text,
    last_name text,
    gps geometry(Point,4326),
    created_at timestamp with time zone NOT NULL DEFAULT now(),
    updated_at timestamp with time zone NOT NULL DEFAULT now(),
    deleted_at timestamp with time zone,
    consented_at timestamp with time zone,
    full_text text GENERATED ALWAYS AS ((((((((((((((((((COALESCE(first_name, ''::text) || ' '::text) || COALESCE(last_name, ''::text)) || ' '::text) || COALESCE(email, ''::citext)::text) || ' '::text)) STORED,
    metadata jsonb GENERATED ALWAYS AS (generate_users_meta(id)) STORED
);
```

**Comment by omohokcoj** [comment]
@arnaud thanks for reporting this bug - it has been fixed in the latest 0.2.33 release (the latest docker image has been updated as well)

**Comment by arnaud** [comment]
Hi @omohokcoj; you did it again!
The `0.2.33` version now works pretty great with updating any entity I throw at it, even the most complex ones with generated columns, auto-calculated one and such.

A big big thank you!
